docs/assumptions.rst

```diff
@@ -1,63 +1,159 @@
+        or by supplying the "write_sync" configuration parameter to UnifyFS
+        on startup, which will cause an implicit "flush" operation after
+        every write (note: the "write_sync" mode can significantly slow down
+        write performance.).
+
+During a write phase, a process can deviate from the bulk synchronous
+I/O pattern and read any byte in
+a file, including remote data that has been written by processes
+executing on remote compute nodes in the job.
+However, the performance will differ based on which process wrote the data that
+is being read:
+      - If the bytes being read were written by the same process that is reading
+        the bytes, UnifyFS offers the fastest performance and no synchronization
+        operations are needed. This kind of access is typical in some I/O
+        libraries, e.g., HDF5, where file metadata may be updated and read by
+        the same process.
+      - If the bytes being read were written by a process executing on the same compute
+        node as the reading process, UnifyFS can offer slightly slower performance
+        than the first case and requires no additional synchronization operations.
+      - If the bytes being read were written by a process executing on a different
+        compute node than the reading process, then the performance is slower
+        than the first two cases and the application must
+        introduce synchronization operations to ensure that the most recent
+        data is read.
+In summary, reading the local data (which has been written by processes
+executing on the same compute node) will always be faster than reading
+remote data.
+
+Note that commit semantics also require synchronization for potentially conflicting
+write accesses. If an application does not enforce sequential ordering of file
+modifications during a write phase, e.g., with MPI synchronization,
+and multiple processes write concurrently to the same file offset or to an
+overlapping region, the result is undefined and may
+reflect the result of any of the processes' operations to that offset or region.
+**I don't think this paragraph is true. I think we won't return the last write even if synchronization is applied**
+
+.. How can users check that their application is correctly synchronized? Will we have the checker scripts ready?
+
+'''''''''''''''''''''''''''
+Lamination Consistency Semantics in UnifyFS
+'''''''''''''''''''''''''''
+
+The other consistency model that UnifyFS employs is called "lamination
+semantics" which is intended to be applied once a file is done being modified
+at the end of a write phase of an application.  After a file is
+laminated, it becomes permanently read-only and its data is accessible across
+all the compute nodes in the job without further synchronization.
+Once a file is laminated, it cannot be further modified,
+except for being renamed or deleted.
+.. Is the next sentence true? Does more need to be added?
```

Comment:
I think we need a separate section on failures, because it really depends on where the failure occurs, regardless of lamination state. If a node goes down, it takes the server with it, and all data from the node is lost. If the application has a terminal failure, and the server is still running, another local or remote client can still access the data. If the server encounters a terminal failure, all data on the node is lost.

Reply:
Thanks @MichaelBrim. I added a new section on failure behavior